frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx

```diff
@@ -30,14 +31,34 @@ export default function CourseDescription() {
   const course = useSelector((state) => state.courses.course);
   const coursesInPlanner = useSelector((state) => state.planner.courses);
   const courseInPlanner = coursesInPlanner.has(id);
+  const planner = useSelector((state) => state.planner);
+  const degree = useSelector((state) => state.degree);
   const [show, setShow] = React.useState(false);
   const [loading, setLoading] = React.useState(false);
   const [pageLoaded, setpageLoaded] = React.useState(false);
+  const [coursesPathTo, setCoursesPathTo] = React.useState({});
+
+const getPathToCoursesById = async (id) => {
+  try {
+    const res = await axios.post(
+      `http://localhost:8000/courses/coursesUnlockedWhenTaken/${id}`,
+      JSON.stringify(prepareUserPayload(degree, planner)),
+        {
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+    );
+    dispatch(setCoursesPathTo(res.data.courses_unlocked_when_taken));
```

Comment:
I don't think this needs to be a `dispatch()` since `coursesPathTo` is local state. 
That's why there's an error in the console:
![image](https://user-images.githubusercontent.com/68191852/157384766-4b94ac7e-0f58-45a7-9cda-5eaebe74c9b2.png)

So I think if you should just remove dispatch, everything should work fine. Besides that, everything looks good to me! :)

Reply:
Fixed